grc/gui/ParamWidgets.py

```diff
@@ -17,40 +17,53 @@
 
 from __future__ import absolute_import
 import os
+import configparser
+import subprocess
 
 from gi.repository import Gtk, Gdk
 
+from . import Constants
 from . import Utils
+from .canvas.colors import LIGHT_THEME_STYLES, DARK_THEME_STYLES
 
 
-style_provider = Gtk.CssProvider()
-
-style_provider.load_from_data(b"""
-    #dtype_complex         { background-color: #3399FF; }
-    #dtype_real            { background-color: #FF8C69; }
-    #dtype_float           { background-color: #FF8C69; }
-    #dtype_int             { background-color: #00FF99; }
-
-    #dtype_complex_vector  { background-color: #3399AA; }
-    #dtype_real_vector     { background-color: #CC8C69; }
-    #dtype_float_vector    { background-color: #CC8C69; }
-    #dtype_int_vector      { background-color: #00CC99; }
-
-    #dtype_bool            { background-color: #00FF99; }
-    #dtype_hex             { background-color: #00FF99; }
-    #dtype_string          { background-color: #CC66CC; }
-    #dtype_id              { background-color: #DDDDDD; }
-    #dtype_stream_id       { background-color: #DDDDDD; }
-    #dtype_raw             { background-color: #FFFFFF; }
-
-    #enum_custom           { background-color: #EEEEEE; }
-""")
-
-Gtk.StyleContext.add_provider_for_screen(
-    Gdk.Screen.get_default(),
-    style_provider,
-    Gtk.STYLE_PROVIDER_PRIORITY_APPLICATION
-)
+def have_dark_theme():
+    """
+    Returns true if the currently selected theme is a dark one.
+    """
+    def is_dark_theme(theme_name):
+        """
+        Check if a theme is dark based on its name.
+        """
+        return theme_name in Constants.GTK_DARK_THEMES or "dark" in theme_name.lower()
+    # GoGoGo
+    config = configparser.ConfigParser()
+    config.read(os.path.expanduser(Constants.GTK_SETTINGS_INI_PATH))
+    prefer_dark = config.get(
+        'Settings', Constants.GTK_INI_PREFER_DARK_KEY, fallback=None)
+    if prefer_dark in ('1', 'yes', 'true', 'on'):
+        theme_name = config.get(
+            'Settings', Constants.GTK_INI_THEME_NAME_KEY, fallback=None)
+        return is_dark_theme(theme_name)
+    theme = subprocess.check_output(
+        ["gsettings", "get", "org.gnome.desktop.interface", "gtk-theme"]
+    ).decode('utf-8').strip().replace("'", "")
+    return is_dark_theme(theme)
```

Comment:
Not working on KDE Plasma (5.20.5) when GTK theme is set to `Breeze` and color scheme set to `Breeze Dark`.

![image](https://user-images.githubusercontent.com/31200881/104962544-9b38c700-5a13-11eb-935a-f21a7791b3e9.png)

In Plasma desktop, theme `Breeze` for GTK [adjusts the color following Plasma color settings](https://www.reddit.com/r/kde/comments/h8sqmg/kde_519_breezedark_missing_gtk3_arch/), so there is no such a theme like `Breeze-dark`. Below is my `~/.config/gtk-3.0/settings.ini`:

```ini
[Settings]
gtk-application-prefer-dark-theme=true
gtk-button-images=true
gtk-cursor-theme-name=breeze_cursors
gtk-cursor-theme-size=24
gtk-decoration-layout=icon:minimize,maximize,close
gtk-enable-animations=true
gtk-enable-event-sounds=1
gtk-enable-input-feedback-sounds=1
gtk-fallback-icon-theme=breeze
gtk-font-name=Noto Sans,  10
gtk-icon-theme-name=breeze-dark
gtk-menu-images=true
gtk-modules=colorreload-gtk-module:window-decorations-gtk-module:appmenu-gtk-module
gtk-primary-button-warps-slider=false
gtk-shell-shows-menubar=1
gtk-theme-name=Breeze
gtk-toolbar-icon-size=GTK_ICON_SIZE_LARGE_TOOLBAR
gtk-toolbar-style=3
gtk-xft-antialias=1
gtk-xft-hinting=1
gtk-xft-hintstyle=hintfull
```

The `gtk-theme-name` field doesn't contain a `dark` pattern, which leads to failing to detect dark theme.

I investigated this part of the code. It seems like even if `gtk-application-prefer-dark-theme` is set to true, GRC checks the theme name. I think this is an unintentional mistake, because if it is intentional, the `gtk-application-prefer-dark-theme` field will never affect the result.

My suggestion is to change the first `return is_dark_theme(theme_name)` to `return True`:

```diff
-        return is_dark_theme(theme_name)
+        return True
```